---
title: "The Genetic Underlyning of Brain Volume Diversity in Fruit Fly"
author: "Mohanad Hussein"
date: "10.10.2023"
output:
 pdf_document:
  toc: true
 html_document:
  toc: true
  toc_float:
   collapsed: true
   toc_depth: 2
 word_document:
  toc: true
bibliography: EPFL report.bib
link-citations: true
---

\pagebreak

# ABSTRACT

Human brain exhibits remarkable morphological variation that is deemed to be correlated with its functional capacity. Dipterans are not so different in that manner with the fruit fly (*Drosophila Melanogaster*) being the mirror for many human traits and has been modeled for wide range of the most challenging diseases like cancer and neurodegenerative diseases. But most profitably, they are extremely convenient for studying evolutionary traits due to the ease of performing mutagenesis and selective breeding. Since conducting brain studies on humans is no close to feasibility because of high complexity of biological functions and hard-to-track genetic diversity, we utilize a subset (75 lines) of the isogeneic and inbred Drosophila Genetic Reference Panel (DGRP) consisting of around 200 homozygous lines that are fully sequenced and annotated to perform Genome Wide Association Study (GWAS) aiming to detect the genetic basis of variation in brain size between DGRP lines. A set of bioinformatics tools are used throughout the analysis and interpretation process, including but not limited to GOWINDA, GeneMania, Rrvigo, Expression Atlas, FlyBase and Docker. We concluded that, for the same set of lines, females have more genetic components contributing to the phenotype than males. A group of genes was spotted such as *for*, *pdm3*, *prel*, *shot*, *retn*, *cher*, *side*, *lva*, and *vap-33a*. A gene function prediction analysis revealed envolved contribution of those genes to functions like neuron projection guidance, axogenesis, dendrite development, axon guidance, regulation of neuron projection development and regulation of dendrite development. These findings drive furthering the analysis to functional validation and incorporating the remaining non-analysed DGRP lines for more comprehensive results especially in male genotypes.

# INTRODUCTION

Brain development has been of the greatest attraction within evolutionary biology. This complex machinery with no less than 86 billion neurons [@herculano-houzel_human_2009] does not only imply the profusion of functions carried out by this neurobiological and cellular networks, but also extremely sophisticated biophysical processes of the neural spike underlying the yet-to-be-resolved human behavior. When looking at the contemporary brain structure, it appealingly incites looking deeper into the evolutionary history of such highly structured organ. The impact of positive selection has been widely investigated as to be a critical driver of brain functional and morphological evolution [@hill_molecular_2005]. With Genome-Wide Association Studies (GWAS) being the main statistical tool to interrogate the genetic component in brain development, a high number of studies explored the structure-function link in brain. As a result, together with the advancements in Magnetic Resonance Imaging (MRI) and Computed Tomography (CT) scanning, brain volume is showed to be positively correlated with intelligence in humans [@jansen_genome-wide_2020]. Furthermore, twin studies using MRI pointed out possibly impactful genes and environmental factors. Yet it fails to pinpoint genes responsible for such variation due to small sample size and ancestral diversity in more complex scenarios [@peper_genetic_2007]. Acknowledging the burden and hurdle of implementing statistically powerful and neurobiologically tractable studies in human beings, it coerces us to utilize model organisms. In this study, we aim to resolve this issue by taking advantage of *Drosophila Melanogaster* with the availability of fully sequenced and annotated genotypes. Even though *Drosophila Melanogaster* is anatomically divergent from humans, some fundamental developmental and molecular pathways are well-maintained across species [@rubin_brief_2000]. Those similarities in genetic circuits -generally known as homologies- are deeply shared between humans and fruit flies in their brain and body structures, and development [@held_deep_2017]. With more advanced molecular, genetic, and neurobiological tools, the fly model stands out for deciphering the complexity of the brain development and functioning but also to model and understand human brain diseases including neurodegeneration [@jeibmann_drosophila_2009]. By collecting a pool of wild-type *Drosophila Melanogaster* females from North Caroline in the USA and full-sibling inbreeding them for 20 consecutive generations, the Drosophila Genetic Reference Panel (DGRP) was established from the genome sequencing of 194 . The DGRP lines present at least 4,672,297 Single Nucleotide Polymorphisms (SNPs), 105,799 polymorphic microsatellites and 36,810 transposable elements, as well as insertion/deletion events and copy number variants that can serve as a resource for understanding genetic architecture of quantitative traits important evolutionarily and modelling human quantitative traits. We derive benefit from the genetic variation between the fully homozygous genotypes in the database and draw a roadmap to pinpoint genes and molecular pathways underlying the variation in size (volume) of *Drosophila Melanogaster* brains. The approach is to collect volumetric data through analyzing micro-CT images of fixed and stained fly heads followed by Genome Wide Association analysis for the brain size trait. The outstanding SNPs are to be curated using a wide range of bioinformatic tools including Gene Set Enrichment Analysis (GSEA) to test the significance and discover the molecular pathways putatively involved in a phenotype (i.e., brain size). The ultimate goal would be to perform functional validation in the candidate genes and perform a proper molecular study to validate the hypothesis which might be taken on later.

# Methods

## Fly preparation and scanning

From a pool of 200 DGRP lines, around 100 lines were bred for 2 generations to eliminate the epigenetic maternal effects and potential environmental factors. The breeding protocol starts by collecting 5 males and 9 females from the main stock to breed and lay eggs for 3 days then removing adults to prevent mating across generations. After 10 days, when the first generation ecloses, a new generation is started by taking 5 males and 9 females to breed for another three days then adults are removed and 12 days are allowed for second generation's eclosure. Finally, one pair (Male-Female) are randomly selected from each line for the pre-scanning preparation. The preparation starts by dissecting fly heads using fine tweezers and fixing head tissues to prevent degradation using Bouin’s solution. Fixation lasts for 16-24 hours and is followed by 3 washes of 30 minutes with micro-CT buffer (0.1M Na2HPO4,1.8% sucrose) then staining with iodine for 7-10 days before scanning. To ensure noise-free fly head scanning, water-filled pipette tips are used to store heads. Gum-sealed from the two ends, the tips are placed in 3D-printed plastic holders for no more than 2 days prior to scanning date to avoid stain diffusion from the brain tissue.

## Computed Tomography (CT) image analysis

Images obtained from CT scanning are processed using DragonFly software [@gendron_centralizing_2021]. The software allows manipulating images by filtering according to projection domains. Noise in the image induced by stain in water surrounding the head can be reduced and intensity of highly stained brain parts can be further magnified to ease detection. To produce a segmented 3D image, anything else but the brain tissue is masked over the layers by reducing signal threshold, and then brain tissue (which signals higher) is segmented (highlighted) to obtain a final image of the brain. When segmenting the whole brain, it is important to first centralize the image in all three axes, this ensures a consistent and parallel (i.e. similarly highlighted from all dimensions) detection of morphologically diverse brain parts with better precision. This allows a standardization of the segmentation procedure and minimizes human error by always keeping the same orientations of the brains. The segmented voxels form a set of values that represent the volume of the brain. The obtained 3D masked images were analysed using Topological Data Analyses to unbiasedly, and in an unsupervised manner, describe the morphology of the brains and the inner anatomical structures.

## Genome Wide Association analysis

We retrieved genomic data from the *Drosophila* *Melanogaster* Genome Reference Panel (DGRP) and incorporated it to perform the GWA analyses using the web-based software DGRPool [@gardeux_dgrpool_2023]. This computational platform is an online tool that implements automated GWAS through PLINK2, an R software and package for genotype/phenotype mapping and data analysis. DGRPool runs a corrected GWAS model that accounts for six covariates (Wolbachia and 5 main inversions) and is based on the dm3 assembly variants available in *Drosophila Melanogaster* Genome Reference Panel 2 (DGRP2). The analysis is done by formatting the files according to the required TSV format that must contain single individual data points for each sex of each line (an example can be found on the tool website). The order of males and females is not important but data points corresponding to lines that do not exist in the sequence database are not taken into account as well as data points for a single sex without the other. The analysis outputs a list of associated variants with a p-value, a multiple hypothesis correction of p-value using Benjamini-Hochberg method, the variant, it’s reference and chromosomal location and the gene/s of which it overlaps with and, most importantly, the effect size of each variant compared to its reference allele.

## Manual variant curation

There are some conditions under which GWAS results are not comprehensive enough (see discussion). For such purpose manual curation is required to eliminate false positives, which can be inferred by considering confounding factors like environmental exposure. The pursued pipeline was established through trial and error, accompanied by continuous exploration of databases and online tools. Deciding on a False Discovery Rate (FDR)-corrected p-value up to 0.05 as the maximum cut-off of the variants to be investigated, the data investigation starts by looking up the relevant gene’s expression record in the Single Cell Atlas Expression and the Fly cell atlas project, both offer exoression data for several experimental conditions such as stress or biological ageing [@davie_single-cell_2018, @li_fly_2022]. There exist multiple ways to visualize data depending on the statistical method we use. The commonly used methods are tSNE and UMAP which both perform dimensionality reduction. From here, we investigate the most relevant study and check gene expression data in the fly brain tissue. The study type mainly investigated was genetic, genomic, and computational resource for exploring neural circuit function [@davis_genetic_2020]. The next step is looking up associated phenotypes for the variant as it will be highly under focus if the gene is correlated with neural growth or central nervous system morphology in general. FlyBase is also checked for variant annotation, whether the variant was Another piece of information added to the metadata is the functional annotation of the variant as it can possess a regulatory role as well.

## Gene Set Enrichment Analysis (GSEA) and GOWINDA

The list of variants obtained from a GWAS study can occasionally exceed 100 SNPs which makes it overwhelming to manually curate. In such cases, GSEA is done using GOWINDA to highlight the redundant Gene Ontology (GO) terms by classifying genes as involved in a few or all of three domains: biological process, molecular function and cellular component. This classification paves the road for highlighting the gene ontology category impacting the phenotype undergoing scrutiny. GOWINDA also untangles the violated assumption by GWAS studies, that is, all genes are samples independently and equally [@kofler_gowinda_2012]. While in fact, longer genes typically have more variants and thus are more likely to stand out in sampling. The tool requires 4 datafiles: genome annotation file (in .gft format), two files for total SNPs and filtered SNPs (.txt, normally these are obtained from GWAS with p \<0.05 subset of the data being the target SNPs) and a gene-GO terms mapping file (.txt, obtained from FunctionAssociate2 or Ensemble) [@berriz_next_2009], [@martin_ensembl_2023]. The simulation can be adjusted as stringent by selecting the -gene mode to count genes only once in GO categories, details are available in this project’s Git repository.

## Interpretation of GOWINDA results using rrvgo

The hierarchical structures of gene ontology terms and categories often leads to redundant enrichment analysis results that are challenging to interpret. Like many other tools, rrvgo resolves this issue by reducing terms upon semantic similarity and providing comprehensive graphics and various visualization options [@sayols_rrvgo_2023]. Gene ontology terms and scores obtained from GSEA are directly used in rrvgo together with the fly genome-wide annotation package in Bioconductor [@carlson_orgdmegdb_2017]. Reduced terms are refiltered again for non-zero scores (terms with no 0 genes underlisted) and the following phrases within GO terms (neuron, projection, cell, proliferation, mushroom). Semantically most relevant terms were compared against GWAS results by using Gene Sets Characterizing & Functional Trend Analysis (FuncAssociate 2.0) dataset [@berriz_next_2009].

# RESULTS & DISCUSSION

High-throughput omics experiments often rely on hundreds of simulations to resolve the complexity and tremendous number of interactions that can take place in a biological system. Since it is extremely challenging to model each and every aspect of a biological scenario with high precision and the noise caused by environmental factors can also mislead the search for a causal correlation, multiple simulations uncover a wide range of possible scenarios and reduce the uncertainty of prediction and modelling [@athieniti_guide_2023]. The same logic applies for GWAS as well, with many variants being sampled against the genome. A common post-GWAS strategy is to analyze gene sets to test for association between a phenotype and a set of SNPs. Here is where Gene Ontology comes to play, by grouping genes with common functional definition in terms of biological pathways. [@dessimoz_how_2017]. Considering the hierarchical representation in GO results, a term reduction algorithm (rrvgo) helps narrowing down biological processes involved in brain volume diversity as illustrated in **Figure 1**. In female lines, a list of pathways related to neural processes appear to underlie the variation in brain volume (Figure 1). Pathways like “regulation of neuromuscular junction”, “apoptotic process” and “regulation of cell fate specification” are pathways that are highly enriched in GWAS. This finding builds up for the hypothesis that, unlike the prevailing theory of sexual dimorphism in brain size [@cachero_sexual_2010], the phenotype in question (i.e. brain volume diversity) might be genetically controlled rather that sex-driven. However, some limitations do exist in this approach; the fact that brain size-data gathered represent only a portion of the full DGRP lines (75 out of 200), and that the experiment is done in one replicate only. This raises the challenge of detecting real variation effect in high statistical power. Most importantly, using fewer fly lines reduces the detectability of genetic variation’s effect size if it is already in low magnitude. With the possibility of technical variability occurrence, it is therefore increasingly important to ensure implementation of all lines in the DGRP set. The lack of a gold standard list for gene sets is also a major concern when applying the same analysis using other curated gene lists [@sun_powerful_2019].

<center>

![**Figure 1:** A tree map generated using reduced GO terms. The rrvgo tool is used for GO term reduction. The map shows GO terms of Biological Functions predicted to be impacting brian size in flies. Highlighted in yellow are the two of the three most significant biological processes involved in brain size development. GO terms are ranked upon the their size (i.e. the number of genes listed within the category), bigger squares mean highly ranked terms.](treemap%20F.png)

<center>

The genetic basis of brain volume diversity is a hypothesis yet under development. As the goal here is to highlight brain volume related neural traits, by filtering the reduced matrix of gene ontology terms, the three most significant terms associated with neural traits were selected from the non-zero sized list, these are: neuron axon guidance, regulation of neuron projection development and neuron projection morphogenesis. Many other terms were selected but upon further gene investigation and comparison with GWAS results, they were eliminated due to lack of significance (i.e. genes in GO terms not ranked in the top 5%) or lack of sufficient data inferring effect size as some variants were not reported in more than 3-5 lines. This led to excluding those variants together with the ones having equal or low effect size to that of the reference allele. Since GWAS is performed separately for each sex, the downstream analysis is also done in the same manner. Nevertheless, depending on the phenotype used, GWAS results varied between males and females. Using classical morphometry, which focuses on measurable features to describe the geometric properties of fly brain, we found no significant association of any SNP with brain volume variation in males of 75 DGRP lines. Therefore, results from female GWAS are included here.

<center>

![**Figure 2:** **A)** Effect sizes of variants of genes curated using reduced GO terms, calculated using mean difference of reference allele versus variant. **B)** Gene interaction network for genes in panel A. Interaction analysis done with GeneMania.](variant%20effect%20size_GeneMania.png)

<center>

To put GWAS findings into context and direct how [@kondo_cross-species_2009]this experiment can proceed, we implement a filtering step that aims to find genes of high significance controlling the phenotype. From the enriched and reduced GO terms, we track genes that are in the top 5% relevant SNPs to brain size variation and have a function or more contributing to this trait. In **Figure 2** (A), the scatter plot shows the comparison of effect sizes of SNPs in genes involved in pathways related to brain size variation (phenotype measurements are obtained from DGRPool). In the category of neuron axon guidance, the following genes had variation with high effect sizes: foraging (for) involved in cell size regulation and has a transcription factor binding site (TFBS) for daughterless (da) that participates in regulating neurogenesis, cheerio (cher) and sidestep (side) both have common functions of neuron projection guidance and neuron fate commitment with the latter having an extra function in axonogenesis. Under the category of neuron projection morphogenesis, the highest contribution of genes was observed. Pou domain motif 3 (pdm3) gene contributes to neuron projection guidance and axonogenesis, in addition to dendrite morphogenesis and development. Preli-like (prel) controls autophagy which in role, plays an important role in neurogenesis. Retained (retn) is a transcription factor that positively regulates neuron regulation and differentiation as well as glial cells development. In the last category of GO, regulation of neuron projection development, upstream variation detected in short stop (shot), a gene that contributes to neuron projection guidance, axonogenesis and axon guidance. Two particular genes were redundant in neuron projection morphogenesis and regulation of neuron projection development; lava lamp (lva) that encodes microtubule-associated protein involved in dendrite morphogenesis, regulation of neuron projection and neurogenesis. VAMP-associated protein (vap-33) overlaps with lva and controls synaptic growth. Both genes have three variants on 3’ UTR and downstream of the gene.

Majority of variation exists in intronic regions. This highlights that genomic regions of unknown regulatory annotation might actually be vital for the aforementioned genes’ regulation. Aiming to widen the picture and bring more understanding of how the those genes contribute to variation in brain volume, GeneMania, a biological network integration and gene prioritization tool, is used to create a connection map between genes [@warde-farley_genemania_2010]. The connection network links input genes to each other and to other genes underlying the phenotype. In **Figure 2** (B), the network represents the functions shared by each of the gene in the input list. Neuron projection guidance, axonogenesis and dendrite development are highlighted as the most enriched functions in brain size variation (for more details:[@genemania_gene_2024]).

In GSEA, one of three categories (overrepresentation analysis, functional class scoring, pathway topology-based methods) of enrichment methods is usually implemented (Maleki ref). In this study, overrepresentation analysis is carried out using GOWINDA and rrvgo. Even though the limitations on this study might hinder our ability to pinpoint a causal variant of diversity in brain volume, it provides us with a roadmap to further increase the statistical power of the analysis pipeline. Upon the incorporation of missing DGRP lines in this analysis and replicating the screening process, a plausible scenario to underline the causal genes is to perform functional validation using constructs to study genes in question. For example, rescue experiments could be performed using overexpression and RNAi constructs to overexpress and knockdown genes for the inspection of their role in brain size determination [@kondo_cross-species_2009]. For genes with intronic variation, enhancers/silencer trap constructs are optimal for regions with regulatory functions [@manseau_gal4_1997]. In such a variable phenotype, Quantitative Trait Locus (QTL) mapping can be implemented by crossing two genetically distinct lines to pinpoint linked genetic markers to trait in question [@r.smith2020]

To conclude, the outcome of partial utilization of DGRP lines promotes integrating the remaining lines into the analysis. Promisingly, identifying the genetic basis of brain size variation will lead to deeper understanding of brain development. This will consequently help pinpointing molecular pathways involved in brain growth and maturation. As many neurological diseases are related to brain size alterations, like schizophrenia and microcephaly, it is therefore important to highlight genes that alter brain size and develop diagnostic tools and therapeutic methods if possible.

# DATA AVAILABILITY

All codes, pipelines, data files and additional analysis reports used and obtained in this analysis can be found on my Github repository for this project (<https://github.com/mohanadhussein>).

# ACKNOWLEDGMENT

The completion of this project could not have been possible without the support of Dr. Ana Marija Jakšić, the leader and director of the Jakšić laboratory at the Brain Mind Institute in EPFL, Lausanne. Her continuous support, patience and feedback were instrumental for the successful completion of my summer internship of which this project was a part of. I deeply appreciate providing me with this opportunity which has broadened my horizons in both professional and personal development. A dept of gratitude is also owed to Samuel Bourgeat, the PhD student at the lab, for his primary mentoring and step-by-step assistance. His welcoming and support made me feel part of the team since the first meeting. My gratitude is indeed extended to the remaining group members: Ivan Tomic, Riddha Manna, Carlos Henrique from Brisken group, Lisa Watt, Manon Pribille, Marie-France, Gonçalo Braga and Victoir Gamblin, for every help and moment spent in and outside the lab. Thanks to everyone who contributed to making these three months pass smoothly and in the most productive way possible. I will always cherish the knowledge, the experience, and the friendships I have built during this time. Hopefully, will be back next year for master’s.

# REFERENCES
