-GOWINDA runs Gene Ontology (GO) for high throughput analysis producing extinsive gene lists such as GWAS. The logic and usage of the tool is found on the developer web page: [https://sourceforge.net/p/gowinda/wiki/Main/].\
-The analysis is done separately for each sex.\
-Input files are found in the GOWINDA folder of the project repository and are recognized by the prefix *Abs_vol* (stands for Abslute Volume). The query SNP list is basically a cutoff of the full SNP list obtained from GWAS. As for the Genome index gtf file, since it is a huge file and can't be uploaded to github, downloading it is descriped in [https://wiki.flybase.org/wiki/FlyBase:Downloads_Overview#GTF_files]. The version of the index file needs to match the assembly version of which DGRPool depends on while conducting GWAS in the first place (i.e. Dmel r6).\
-For the sake of learning, Docker was used to build an application for running the analysis on a remote Imac in the lab. The Docker files are also added to the same folder. Noting that Docker is not necessary at all, and the analysis can be run on average mac devices.\
-Output files are recognized by the prefix *results*. The snp/gene annotation in the file's name refers to the mode of variant selection (explained in GOWINDA's tutorial)\
-**Imp note: In docker files, you might see some entries saying, Entropy0 or Entropy1, those were later replaced by Abs_vol analysis done in this analysis, the Entropy analysis will be performed later**
